cmd/state/internal/cmdtree/checkout.go

```diff
@@ -40,12 +40,16 @@ func newCheckoutCommand(prime *primer.Values) *captain.Command {
 				Description: locale.Tl("flag_state_checkout_force", "Leave a failed project checkout on disk; do not delete it"),
 				Value:       &params.Force,
 			},
+			{
+				Name:        "from-archive",
+				Description: locale.Tl("flag_state_checkout_from_archive", "Checkout from the given .tar.gz archive"),
+				Value:       &params.FromArchive,
+			},
```

Comment:
That sounds reasonable, as long as the archive path always has an extension. Otherwise the argument is ambiguous, as it can be either `org/project`, `project`, or `path/to/tar.gz`.

Reply:
Sounds good, I don't see any reason why it wouldn't have an extension.